src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -232,6 +238,46 @@ def normalize_disk_info(image_data_disks=None,
         if isinstance(data_disk_delete_option, str):
             info[i]['deleteOption'] = data_disk_delete_option
 
+    # add copy data disks
+    i = 0
+    source_resource_copy = list(source_snapshots_or_disks)
+    source_resource_copy_size = list(source_snapshots_or_disks_size_gb)
+    while source_resource_copy:
+        while i in used_luns:
+            i += 1
+
+        used_luns.add(i)
```

Comment:
This business logic seems difficult to understand~ Could you please add comments to explain it?

Reply:
It is an existing logic and also has comments
https://github.com/Azure/azure-cli/blob/bb30f28efad520b36342ece7689b3980854a158a/src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py#L244-L259